bakery/authorizer.go

```diff
@@ -0,0 +1,134 @@
+package bakery
+
+import (
+	"golang.org/x/net/context"
+	"gopkg.in/errgo.v1"
+
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
+)
+
+const Everyone = "everyone"
+
+// Authorizer is used to check whether a given user is allowed
+// to perform a set of operations.
+type Authorizer interface {
+	// Allow checks whether the given identity (which will be nil
+	// when there is no authenticated user) is allowed to perform
+	// the given operations. It should return an error only when
+	// some underlying database operation has failed, not when the
+	// user has been denied access.
+	//
+	// On success, each element of allowed holds whether the respective
+	// element of ops has been allowed, and caveats holds any additional
+	// third party caveats that apply.
+	Authorize(ctxt context.Context, id Identity, ops []Op) (allowed []bool, caveats []checkers.Caveat, err error)
```

Comment:
I can't think of one, I was just looking at the consistency. I'm happy to trust your judgment here.